incubator/hnc/api/v1alpha2/hnc_config.go

```diff
@@ -55,18 +55,18 @@ const (
 
 	// Condition reasons for BadConfiguration
 	ReasonMultipleConfigsForType = "MultipleConfigurationsForType"
-	ReasonTypeNotFound           = "TypeNotFound"
+	ReasonResourceNotFound       = "ResourceNotFound"
 
 	// Condition reason for OutOfSync, e.g. errors when creating a reconciler.
 	ReasonUnknown = "Unknown"
 )
 
-// TypeSynchronizationSpec defines the desired synchronization state of a
+// ResourceSpec defines the desired synchronization state of a
 // specific resource.
-type TypeSynchronizationSpec struct {
+type ResourceSpec struct {
 	// Group of the resource defined below. This is used to unambiguously identify
-	// the resource.
-	Group string `json:"group"`
+	// the resource. It may be omitted for core resources (e.g. "secrets").
+	Group string `json:"group,omitempty"`
```

Comment:
Nice, it will turn, e.g.
```
spec:
  group: ""
  resource: secrets
  mode: Propagate
```
into
```
spec:
  resource: secrets
  mode: Propagate
```
right?

Reply:
Yup, exactly